### PR TITLE
Fix bug #2731 on stripping query comments for snowflake

### LINKED ISSUE #2731: EOL SQL comments remove entire line from execution
### Describe the bug
When using SQL line comment (`-- comment`) at the end of lines **and** running multiple SQL queries (separated by semicolons) within one [statement block](https://docs.getdbt.com/reference/dbt-jinja-functions/statement-blocks/), if any **one-line** query has a line comment after it (on the same line), that command won't be executed (i.e. not submitted to the database at all) - except if it's the first query.
 
- This affects all statement blocks, whether they come from a model, macro, materialization, etc.
- Block comments (`/* comment */`) do not have this effect (which provides a workaround)

This will work (all three queries will be executed):
```
{%- call statement('test') -%}
  select 1;
  select 2;
  select 3;
{%- endcall %}
```

This will work (all three queries will be executed):
```
{%- call statement('test') -%}
  select 1; /* comment */
  select 2; /* comment */
  select 3; /* comment */
{%- endcall %}
```

**This will not work** (only the first query will be executed):
```
{%- call statement('test') -%}
  select 1; -- comment
  select 2; -- comment
  select 3; -- comment
{%- endcall %}
```

This will work (all three queries will be executed):
```
{%- call statement('test') -%}
  select
    1; -- comment
  select
    2; -- comment
  select
    3; -- comment
{%- endcall %}
```

### Steps To Reproduce
Just put the examples above in a model or macro, and take a look at the queries executed on your database (query log, etc.)

### Expected behavior
All three queries should be executed in all examples.

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [x] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
installed version: 0.18.0-rc1
   latest version: 0.17.2

Your version of dbt is ahead of the latest release!

Plugins:
  - snowflake: 0.18.0rc1
```

**The operating system you're using:**
```
macOS Catalina
Version 10.15.6 (19G2021)
```

**The output of `python --version`:**
```
Python 3.8.3
```

### PR DESCRIPTION
**Pull request by rvacaru**
resolves #2731

### Description

Improved regex which removes EOL comments from snowflake multiline queries.

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Review by jtcohen6 — COMMENTED** [review]
Thanks for the fix @rvacaru, and for the thorough unit testing!

Do you have a strong feeling about adding those in a new file (`test_connections.py`), versus adding it to the existing tests in [`test_snowflake_adapter.py`](https://github.com/fishtown-analytics/dbt/blob/dev/margaret-mead/test/unit/test_snowflake_adapter.py)? I ask because, someday, we'll want to split up these tests by adapter plugin, and it would be great to keep them closer together.

Finally, could you pull `dev/margaret-mead` and move your Changelog entry up to the `v0.20.0` section (beneath a new `### Fixes` subsection)?

**Comment by rvacaru** [comment]
You're welcome @jtcohen6 ! I moved the unit tests and updated the changelog like suggested, it should be ok now.

File: test/unit/test_snowflake_adapter.py [hunk 1]
```diff
@@ -523,3 +524,42 @@ def test_float_from_description(self):
...
+
+class SnowflakeConnectionsTest(unittest.TestCase):
+    def test_comment_stripping_regex(self):
+        pattern = r'(\".*?\"|\'.*?\')|(/\*.*?\*/|--[^\r\n]*$)'
+        comment1 = '-- just comment'
+        comment2 = '/* just comment */'
+        query1 = 'select 1; -- comment'
+        query2 = 'select 1; /* comment */'
+        query3 = 'select 1; -- comment\nselect 2; /* comment */ '
+        query4 = 'select \n1; -- comment\nselect \n2; /* comment */ '
```
Comment (jtcohen6) [inline test/unit/test_snowflake_adapter.py:537]: Could you add one more unit test that represents the original error case reported in #2731:
```
query5 = 'select 1; -- comment \nselect 2; -- comment \nselect 3; -- comment'
expected_query_5 = 'select 1; \nselect 2; \nselect 3;'
```

**Review by jtcohen6 — COMMENTED** [review]
Thanks @rvacaru! This is looking good.

I recommended one more unit test, for good measure, to be extra sure we're resolving the initially stated issue.

In testing this out I realized that there's a slight inconsistency arounnd query splitting and query headers, in that the latter only prepend the first split query:
```
2021-01-20 08:09:01.951967 (Thread-1): On model.testy.a: /* {"app": "dbt", "dbt_version": "0.19.0rc2", "profile_name": "jaffle_shop", "target_name": "dev", "node_id": "model.testy.a"} */
select 1; -- comment
2021-01-20 08:09:02.190950 (Thread-1): SQL status: SUCCESS 1 in 0.24 seconds
2021-01-20 08:09:02.191158 (Thread-1): select 2;
2021-01-20 08:09:02.191315 (Thread-1): Using snowflake connection "model.testy.a".
2021-01-20 08:09:02.191677 (Thread-1): On model.testy.a: select 2; -- comment
2021-01-20 08:09:02.554744 (Thread-1): SQL status: SUCCESS 1 in 0.36 seconds
2021-01-20 08:09:02.554959 (Thread-1): select 3;
```

That's not in scope for this PR, I just wanted to get it down in writing in case someone cares to open an issue later on.

File: test/unit/test_snowflake_adapter.py (hunk 1 above)
Comment (rvacaru) [reply test/unit/test_snowflake_adapter.py:537]: I've added the extra test. Let me know if there's anything more

**Review by jtcohen6 — APPROVED** [review]
@rvacaru Thanks for the contribution!

**Pull request merged**